libcontainerd/remote/client.go

```diff
@@ -571,7 +571,7 @@ func closeStdin(ctx context.Context, p containerd.Process) error {
 
 func (c *client) processEvent(ctx context.Context, et libcontainerdtypes.EventType, ei libcontainerdtypes.EventInfo) {
 	c.eventQ.Append(ei.ContainerID, func() {
-		err := c.backend.ProcessEvent(ei.ContainerID, et, ei)
+		err := c.backend.ProcessEvent(ctx, ei.ContainerID, et, ei)
```

Comment:
Wondering about this one too. In principle I think it should also be not-cancellable, but looking at the code, the context here is "the main context" that's tied to the lifetime of the daemon and it will be cancelled when `DaemonCli.start` finishes. But at this point, the whole program has little time left before the main goroutine exits, so the goroutine calling these functions may not have a chance to run this anyway.

Reply:
Would love to hear others thoughts on this.
All these queue processing contexts would be inheriting from the daemon start context.
Does it make sense to try and process events we've already received before letting the daemon shut down?

I think I assumed for all of these the inherited context would only ever be cancelled by daemon shutdown.